portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java

```diff
@@ -2771,6 +2777,82 @@ private long _getValidFileEntryTypeId(
 		}
 	}
 
+	private void _inheritRolesPermissions(
+			long inheritableParentFolderId, DLFileEntry dlFileEntry)
+		throws PortalException {
+
+		long companyId = dlFileEntry.getCompanyId();
+		String parentClassName = DLFolderConstants.getClassName();
+
+		if (inheritableParentFolderId ==
+				DLFolderConstants.DEFAULT_PARENT_FOLDER_ID) {
+
+			int count =
+				_resourcePermissionLocalService.getResourcePermissionsCount(
+					companyId, DLConstants.RESOURCE_NAME,
+					ResourceConstants.SCOPE_INDIVIDUAL,
+					String.valueOf(dlFileEntry.getGroupId()));
+
+			if (count == 0) {
+				_resourceLocalService.addResources(
+					companyId, dlFileEntry.getGroupId(), 0,
+					DLConstants.RESOURCE_NAME,
+					String.valueOf(dlFileEntry.getGroupId()), false, true,
+					true);
+			}
+
+			parentClassName = DLConstants.RESOURCE_NAME;
+			inheritableParentFolderId = dlFileEntry.getGroupId();
+		}
+
+		Map<Long, Set<String>> dlFolderRoleIdsToActionIds =
+			_resourcePermissionLocalService.
+				getAvailableResourcePermissionActionIds(
+					companyId, parentClassName,
+					ResourceConstants.SCOPE_INDIVIDUAL,
+					String.valueOf(inheritableParentFolderId),
+					ListUtil.fromCollection(
+						ResourceActionsUtil.getModelResourceActions(
+							DLFolderConstants.getClassName())));
```

Comment:
ResourceActionsUtil.getModelResourceActions() returns a list. No need to use ListUtil.fromCollection() to convert it to a list.

Reply:
Fixed it. Thanks